### PR TITLE
Split reusable part of build&test workflow into a separate file, call it in various modes from other files

### LINKED ISSUE #1357: Refactor build and test workflow to allow flexible reuse for push/commit, schedule runs on different runners and different configurations
| | Push/commit | Schedule | On demand |
|--------|--------|--------|--------|
| Rolling driver runner and forked pytorch+IPEX | Python 3.9, 3.10, 3.11 for llvm-target, 3.9 for other branches | No | Yes |
| Rolling driver runner and upstream pytorch, no IPEX | No | Yes (python versions?) | Yes |
| LTS driver runner and forked pytorch+IPEX | Python 3.9, 3.10, 3.11 for llvm-target, 3.9 for other branches | No | Yes | 
| LTS driver runner and upstream pytorch, no IPEX | No | Yes (python versions?) | Yes | 
| A770 runner (rolling driver) and forked pytorch+IPEX | No | Yes (python versions?) | Yes | 
| A770 runner (rolling driver) and upstream pytorch, no IPEX | No | Yes (python versions?) | Yes | 

Parameterized workflow inputs:
1. Runner label
2. Pytorch flavor (ipex/no ipex)
3. Python version

### PR DESCRIPTION
**Pull request by gshimansky**
Fixes #1357 

The core Triton is a small number of people, and we receive many PRs (thank
you!).  To help us review your code more quickly, **if you are a new
contributor (less than 3 PRs merged) we ask that you complete the following
tasks and include the filled-out checklist in your PR description.**

Complete the following tasks before sending your PR, and replace `[ ]` with
`[x]` to indicate you have done them.

- [x] I am not making a trivial change, such as fixing a typo in a comment.

- [x] I have written a PR description following these
  [rules](https://cbea.ms/git-commit/#why-not-how).

- [ ] I have run `pre-commit run --from-ref origin/main --to-ref HEAD`.

- Select one of the following.
  - [ ] I have added tests.
    - `/test` for `lit` tests
    - `/unittest` for C++ tests
    - `/python/test` for end-to-end tests
  - [x] This PR does not need a test because `this PR doesn't change any triton code`.

- Select one of the following.
  - [x] I have not added any `lit` tests.
  - [ ] The `lit` tests I have added follow these [best practices](https://mlir.llvm.org/getting_started/TestingGuide/#filecheck-best-practices),
    including the "tests should be minimal" section. (Usually running Python code
    and using the instructions it generates is not minimal.)

### CODE REVIEW

**Comment by pbchekin** [comment]
Didn't run on PR commit?

**Comment by gshimansky** [comment]
> Didn't run on PR commit?

Yes, on PR inputs are empty and it broke all boolean parameters without any diagnostics from GA.

**Review by pbchekin — APPROVED** [review]
Looks good!
To merge this PR we will need to update our branch protection rules.

File: .github/workflows/build-test-reusable.yml [hunk 1]
```diff
@@ -0,0 +1,259 @@
...
+          ref: ${{ env.TRANSFORMERS_VERSION }}
+          try-tag-prefix: v
+
+      - name: Run E2E test
+        run: |
+          # Set WORKSPACE for inductor_xpu_test.sh to make sure it creates "inductor_log" outside of pytorch cloned directory
+          export WORKSPACE=$GITHUB_WORKSPACE
+          cd pytorch
+          pip install pyyaml pandas scipy numpy psutil pyre_extensions torchrec
+          # TODO: Find the fastest Hugging Face model
```
Comment (leshikus) [inline .github/workflows/build-test-reusable.yml:228]: this is not a good place for this TODO, the tests come from developers, and they do not read our workflows; I believe it's better to file it as issue and assign to @etiotto

**Review by leshikus — COMMENTED** [review]
Looks good

**Pull request merged**